src/cow-react/modules/account/containers/Transaction/ActivityDetails.tsx

```diff
@@ -164,6 +168,21 @@ export function ActivityDetails(props: {
   const showProgressBar = (activityState === 'open' || activityState === 'filled') && order?.class !== 'limit'
   const showCancellationModal = activityDerivedState.order ? getShowCancellationModal(activityDerivedState.order) : null
 
+  const { surplusAmount, surplusToken } = useMemo(() => {
+    const output: { surplusToken?: Token; surplusAmount?: CurrencyAmount<Token> } = {}
```

Comment:
We cant do that in this case because we have `useCoingeckoUsdValue` hook that takes `surplusAmount` as a parameter and hook can't be called conditionally. So we cant have `if` or ternary operator before the coingecko hook